src/commonTest/kotlin/kotools/types/collection/NotEmptyMapTest.kt

```diff
@@ -1,15 +1,77 @@
 package kotools.types.collection
 
-import kotlinx.serialization.*
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerializationException
 import kotlinx.serialization.builtins.MapSerializer
 import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
 import kotlinx.serialization.json.Json
 import kotools.types.contentShouldEqual
 import kotools.types.shouldHaveAMessage
 import kotlin.random.Random
 import kotlin.test.*
 
 class NotEmptyMapTest {
+    @Test
+    fun notEmptyMapOf_should_pass() {
+        val head: Pair<String, Int> = "a" to Random.nextInt()
+        val tail: Array<Pair<String, Int>> = arrayOf(
+            "b" to Random.nextInt(),
+            "c" to Random.nextInt()
+        )
+        assertEquals(
+            actual = notEmptyMapOf(head, *tail).toMap(),
+            expected = mapOf(head, *tail)
+        )
+    }
+
+    @Test
+    fun map_toNotEmptyMap_should_pass_with_a_not_empty_Map() {
+        val map: Map<Char, Int> = mapOf('a' to Random.nextInt())
+        val result: Result<NotEmptyMap<Char, Int>> = map.toNotEmptyMap()
+        result.getOrThrow()
+            .toMap()
+            .entries contentShouldEqual map.entries
+    }
+
+    @Test
+    fun map_toNotEmptyMap_should_fail_with_an_empty_Map() {
+        val map: Map<Char, Int> = emptyMap()
+        val result: Result<NotEmptyMap<Char, Int>> = map.toNotEmptyMap()
+        val error: IllegalArgumentException =
+            assertFailsWith(block = result::getOrThrow)
+        error.shouldHaveAMessage()
+    }
+
+    @Test
+    fun head_should_return_its_first_entry() {
+        val expected: Pair<Char, Int> = 'a' to 1
+        val entries: NotEmptyMap<Char, Int> = notEmptyMapOf(expected, 'b' to 2)
+        val head: Pair<Char, Int> = entries.head
+        assertEquals(expected, actual = head)
+    }
+
+    @Test
+    fun tail_should_return_all_its_entries_except_the_first_one() {
+        val expected: NotEmptyMap<Char, Int> = notEmptyMapOf('b' to 2)
+        val entries: NotEmptyMap<Char, Int> = expected.entries.toSet()
+            .map { it.toPair() }
+            .toTypedArray()
+            .let { notEmptyMapOf('a' to 1, *it) }
+        val tail: NotEmptyMap<Char, Int>? = entries.tail
+        val actual: NotEmptyMap<Char, Int> = assertNotNull(tail)
+        assertEquals(expected, actual)
+    }
```

Comment:
Refactor this test by resolving the following tasks:
- Rename the `entries` property as `map` and the `tail` property as `result` for better readability.
- Use the assertions available in the [`kotools.types.Assertions.kt`](https://github.com/kotools/types/blob/5d1dc8f98f2220e96081381809cde9d261d04f8f/src/commonTest/kotlin/kotools/types/Assertions.kt) file.

```suggestion
    @Test
    fun tail_should_return_all_its_entries_except_the_first_one() {
        val expected: NotEmptyMap<Char, Int> = notEmptyMapOf('b' to 2)
        val map: NotEmptyMap<Char, Int> = expected.entries.toSet()
            .map { it.toPair() }
            .toTypedArray()
            .let { notEmptyMapOf('a' to 1, *it) }
        val result: NotEmptyMap<Char, Int>? = entries.tail
        result.shouldBeNotNull() shouldEqual expected
    }
```